src/callback-invocation.h

```diff
@@ -44,8 +44,17 @@ struct callback_invocation_holder
 
     ~callback_invocation_holder()
     {
-        if( invocation )
-            owner->deallocate( invocation );
+        if (invocation)
+        {
+            try
+            {
+                owner->deallocate(invocation);
+            }
+            catch (...)
```

Comment:
Printing this to log doesn't give us much information.
Catch `const std::exception & e` and add `e.what()` to the log.

Reply:
Try/catch changed